Added support for complexer assets

RemyLivewall opened the pull request:
I had some issues with the thumbnail generating in a project I am working on. I changed the logic to use a composition derived from the asset. I also added using the smallest sized video for the generator because when there are multiple video tracks with different sizes some thumbnails come out with the wrong size. I also moved the max duration var to the super class to be able to use that in the same way.

I needed these changes because I am stitching videos together (with the passthrough preset) and rendering the resulting thumbnails. I though the changes might be usable change for the original repo.

HHK1 commented:
Hi @RemyLivewall, thanks for your contribution.
There is an issue with the way you calculate your render size. I have some simple AVAsset videos that now generate the thumbnails with an incorrect size. (see the screenshot attached).
Could you take a look at this please ? It may be because the orientation is not taken into account.

Also, could you please add some code in the example app to test this with assets with multiple video tracks ?

![IMG_1737](https://user-images.githubusercontent.com/6484356/60675814-8d494280-9e7d-11e9-95c4-4fd5f65fc1e2.PNG)

RemyLivewall commented:
I changed the way I work with my videos and fixed a silly bug that I had which resulted in the changes I made to the library. I accidentally added all the video's on top of each other instead of behind each other. The original library works fine besides a few minor field changes.

I will update my pull request with just the field changes I am using now.
